doc/design/python/user_interface.md

```diff
@@ -0,0 +1,358 @@
+# User Interface Design
+
+## Basic Concepts
+### Variable
+A `Variable` represents shared, persistent state manipulated by a Paddle model program.
+
+Variables are maintained by `pd.Variable` class,
+each `pd.Variable` represents a tensor whose value can be changed by running ops on it.
+
+A basic way to create a variable is:
+
+```python
+import paddle as pd
+
+v = pd.Variable(shape=[20, 20])
+```
+
+To make it more converient to share a variable, each `pd.Variable` has a name, 
+one can use a name to get or create a `pd.Variable` by calling `pd.get_variable`, for example:
+
+```python
+# same as 
+v = pd.get_variable(name="v", shape=[20, 20])
+```
+
+By default, Variables are model parameters, and will be updated after the network's back propagation.
+
+One can freeze a variable by setting `trainable` to `False` like:
+
+```python
+v = pd.Variable(shape=[20,20], trainable=False)
+```
+
+Some initizlization strategies may be applied to variables, for example, we may set a variable to zero or gaussian random.
+
+```
+v = pd.Variable(shape=[20,20], initializer=pd.zero_initializer())
+z = pd.Variable(shape=[20,20], initializer=pd.gaussian_initializer(mean=0., std=0.1))
+```
+
+to get the value of the variable, one can call
+
+```python
+print v.val()
```

Comment:
maybe eval() is better

Reply:
this method just return a python numpy value, will not execute a sub-graph.

variable.eval() will be a wrapper based on `pd.eval`, currently is not necessary.